### PR TITLE
refactor: move backup code to backup-format package

### PR DESCRIPTION
**Pull request by Jack-Works**
Notable changes:

## Web:

- All backup-related code moved to `@masknet/backup-format`, old code has been removed.
- Now only `NormalizedBackup` is exposed, the internal format IS NOW OPAQUE, DO NOT rely on the internal structure of it.

| Old name                                                              | New name                                | Note                 |
|-----------------------------------------------------------------------|-----------------------------------------|----------------------|
| ========== Backup APIs                                                |                                         |                      |
| generateBackupJSON                                                    | mobile_generateBackupJSON<br  />mobile_generateBackupJSONOnlyForPersona | Only for mobile now. |
| createBackupUrl                                                       |                                         | Removed              |
| ========== Restore APIs                                               |                                         |                      |
| restoreFromBase64                                                     | mobile_restoreFromBase64                | Only for mobile now. |
| restoreBackup                                                         | restoreUnconfirmedBackup                |                      |
| parseBackupStr                                                        | addUnconfirmedBackup                    |                      |
| checkPermissionsAndRestore<br>checkPermissionAndOpenWalletRecovery    | restoreUnconfirmedBackup                |                      |
| restoreFromObject<br>restoreFromBackup<br>patchCreateOrUpdateRelation<br /> patchCreateNewRelation |                                         | Removed              |

Current APIs:

### Create backup

- `mobile_generateBackupJSON`
- `mobile_generateBackupJSONOnlyForPersona`
- `generateBackupPreviewInfo`
- `createBackupFile`

### Restore backup

- `getUnconfirmedBackup`
- `addUnconfirmedBackup`
- `restoreUnconfirmedBackup`
- `mobile_restoreFromBase64`

## Mobile API:

- `protectPrivateKey` option in `update_persona` is no longer used.
- `persona_restoreFromJson` is deprecated. It's an alias of `settings_restoreBackup`
- `settings_getBackupPreviewInfo` no longer return `undefined`. In case it is not able to generate a backup preview, it throws (old behavior also throws in some cases).
- `BackupPreview` now have 2 new fields: `relations` and `createdAt`

### CODE REVIEW

File: packages/backup-format/src/version-0/index.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { isObjectLike } from 'lodash-unified'
+import { None, Some } from 'ts-results'
+import { createEmptyNormalizedBackup } from '../normalize'
+import type { NormalizedBackup } from '../normalize/type'
+
+export function isBackupVersion0(obj: unknown): obj is BackupJSONFileVersion0 {
+    if (!isObjectLike(obj)) return false
+    try {
+        const data: BackupJSONFileVersion0 = obj as any
+        if (!data.local || !data.key || !data.key.key || !data.key.key.privateKey || !data.key.key.publicKey)
```
Comment (septs) [inline packages/backup-format/src/version-0/index.ts:21]:
```js
['local', 'key.key.privateKey', 'key.key.publicKey'].every((p) => lodash.has(obj, p))
```

File: packages/backup-format/src/version-2/index.ts [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+import { Err, None, Some } from 'ts-results'
+import { createEmptyNormalizedBackup } from '../normalize'
+import type { NormalizedBackup } from '../normalize/type'
+
+export function isBackupVersion2(item: unknown): item is BackupJSONFileVersion2 {
+    try {
+        const x = item as BackupJSONFileVersion2
+        return x._meta_.version === 2
+    } catch {}
+    return false
```
Comment (septs) [inline packages/backup-format/src/version-2/index.ts:23]:
```js
return lodash.get(item, '_meta_.version') === 2
```

File: packages/backup-format/src/version-0/index.ts (hunk 1 above)
Comment (Jack-Works) [reply packages/backup-format/src/version-0/index.ts:21]: I prefer plain javascript 👀

**Comment by wenluomask** [comment]
Please check in Jira: https://mask.atlassian.net/browse/MF-331

**Comment by wenluomask** [comment]
Refer to Jira https://mask.atlassian.net/browse/MF-331

**Pull request merged**